src/orquestra/sdk/_base/cli/_dorq/_workflow/_view.py

```diff
@@ -37,26 +38,30 @@ def __init__(
         self._config_resolver = config_resolver or _arg_resolvers.WFConfigResolver(
             wf_run_repo=wf_run_repo
         )
-        self._wf_run_resolver = wf_run_resolver or _arg_resolvers.WFRunResolver(
+        self._wf_run_resolver = wf_run_resolver or _arg_resolvers.WFRunIDResolver(
             wf_run_repo=wf_run_repo
         )
 
         # text IO
-        self._presenter = presenter
+        self._wf_run_presenter = wf_run_presenter
+        self._error_presenter = error_presenter
 
-    def on_cmd_call(
-        self, wf_run_id: t.Optional[WorkflowRunId], config: t.Optional[ConfigName]
-    ):
+    def on_cmd_call(self, *args, **kwargs):
         try:
-            self._on_cmd_call_with_exceptions(wf_run_id=wf_run_id, config=config)
+            self._on_cmd_call_with_exceptions(*args, **kwargs)
         except Exception as e:
-            self._presenter.show_error(e)
+            self._error_presenter.show_error(e)
 
     def _on_cmd_call_with_exceptions(
         self, wf_run_id: t.Optional[WorkflowRunId], config: t.Optional[ConfigName]
     ):
         # The order of resolving config and run ID is important. It dictactes the flow
         # user sees, and possible choices in the prompts.
         resolved_config = self._config_resolver.resolve(wf_run_id, config)
-        resolved_run = self._wf_run_resolver.resolve_run(wf_run_id, resolved_config)
-        self._presenter.show_wf_run(resolved_run)
+        resolved_run_id = self._wf_run_resolver.resolve(wf_run_id, resolved_config)
+
+        summary = self._wf_run_repo.get_wf_run_summary(
+            wf_run_id=resolved_run_id, config_name=resolved_config
+        )
+
+        self._wf_run_presenter.show_wf_run(summary)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Did you mean to get the workflow run twice?
`self._wf_run_resolver.resolve` will get the status model and return the ID
`self._wf_run_repo.get_wf_run_summary` will then get the status model again
This is why we had `resolve_run`, which returns the model directly (instead of ID) so we don't have to get it from the runtime again.